File: Algorithm-Templates/LeetCode-Daily/0001-TwoSum/TwoSum.c
#include <stdio.h>
#include <stdlib.h>

int* twoSum(int* nums, int numsSize, int target, int* returnSize) {
    int* result = (int*)malloc(sizeof(int) * 2);
    for(int i = 0; i < numsSize; i++) {
        for(int j = i + 1; j < numsSize; j++){
            if (nums[i] + nums[j] == target) {
                result[0] = i;
                result[1] = j;

                *returnSize = 2;
                return result; 
            }
        }
    }

    //not found
    *returnSize = 0;
    return NULL;
}

int main() {
    int nums[] = {2, 7, 11, 15};
    int target = 9;
    int returnSize;
    
    int *result = twoSum(nums, 4, target, &returnSize);

    if (result != NULL && returnSize == 2) {
        printf("[%d,%d]\n", result[0], result[1]);
    } else {
        printf("NOT FOUND\n");
    }

    free(result);
    return 0;
}

File: Algorithm-Templates/LeetCode-Daily/0001-TwoSum/TwoSum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        for i in range(n):
            for j in range(i + 1, n):
                if nums[i] + nums[j] == target:
                    return [i, j]
        return []

if __name__ == "__main__":
    sol = Solution()
    test_nums = [2, 7, 11, 15]
    test_target = 9

    result = sol.twoSum(test_nums, test_target)
    print(f"输入: nums = {test_nums}, target = {test_target}")
    print(f"输出下标: {result}")
